ServiceApp/src/org/droidplanner/services/android/core/drone/manager/MavLinkDroneManager.java

```diff
@@ -311,25 +341,41 @@ protected boolean executeAsyncAction(Action action, ICommandListener listener) {
         }
     }
 
-    private void enableFollowMe(FollowType followType, ICommandListener listener) {
+    private void enableFollowMe(FollowType followType, FollowLocationSource source, ICommandListener listener) {
+        Timber.d("enableFollowMe(): followType=%s source=%s", followType, source);
+
         FollowAlgorithm.FollowModes selectedMode = CommonApiUtils.followTypeToMode(drone, followType);
 
         if (selectedMode != null) {
-            if (followMe == null)
+            if (followMe == null) {
+                Timber.d("enableFollowMe(): followMe is null");
                 return;
+            }
 
-            if (!followMe.isEnabled())
-                followMe.toggleFollowMeState();
+            Timber.d("CURRENT: followMe.enabled=%s followMe.state=%s", followMe.isEnabled(), followMe.getState());
+
+            if (!followMe.isEnabled()) {
+                followMe.toggleFollowMeState(source);
+            } else {
+                followMe.setLocationSource(source);
+            }
 
             FollowAlgorithm currentAlg = followMe.getFollowAlgorithm();
             if (currentAlg.getType() != selectedMode) {
                 if (selectedMode == FollowAlgorithm.FollowModes.SOLO_SHOT &&
-                        !SoloApiUtils.isSoloLinkFeatureAvailable(drone, listener))
+                        !SoloApiUtils.isSoloLinkFeatureAvailable(drone, listener)) {
+                    Timber.w("FollowType is SOLO_SHOT, but SoloLink is not available.");
                     return;
+                }
 
-                followMe.setAlgorithm(selectedMode.getAlgorithmType(this, handler));
+                FollowAlgorithm algo = selectedMode.getAlgorithmType(this, handler);
+                Timber.d("Setting followAlgorithm to %s", algo);
+                followMe.setAlgorithm(algo);
+                followMe.setLocationSource(source);
```

Comment:
@kellyschrock why setting the location source here, since you're already setting it above?